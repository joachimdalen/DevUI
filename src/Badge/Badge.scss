.cb-badge {
  padding: 2px 10px;
  border: 2px solid;
  border-color: #28da28;
  color: #28da28;
  font-weight: bold;
  text-transform: uppercase;
  font-size: 10px;
  
  border-radius: 2px;
}
