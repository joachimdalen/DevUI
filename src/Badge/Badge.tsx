import * as React from "react";
import cx from "classnames";
export interface Props {
  /** this dictates what the button will say  */
  label: string;
}

export class Badge extends React.Component<Props> {
  public render() {
    const { label } = this.props;
    return <span className={cx("cb-badge")}>{label}</span>;
  }
}
