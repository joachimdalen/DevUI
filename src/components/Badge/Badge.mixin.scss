@mixin badge-variant($parent, $background) {
  &#{$parent} {
    color: colorFromBackground($background);
    background-color: $background;
    border: none;
  }
}
@mixin badge-neon-variant($parent, $background) {
  &#{$parent} {
    color: colorFromBackground(lighten($background, 40%));
    border-color: lighten($background, 5%);
    background-color: lighten($background, 40%);
  }
}
