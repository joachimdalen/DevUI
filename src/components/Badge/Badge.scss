@import "./Badge.mixin.scss";

%badge-base {
  background-color: $color-badge-primary;
  border-radius: 2px;
  display: inline-flex;
  margin: 2px;
  box-sizing: border-box;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.#{$provider-prefix}-badge-wrapper {
  @extend %badge-base;
  .#{$provider-prefix}-badge-dismissible {
    align-items: center;
    border-radius: 2px;
    display: flex;
    padding-left: 4px;
    padding-right: 4px;
    box-sizing: border-box;
    font-size: 12px;
    &:hover {
      background-color: lighten($color-danger, 30%);
      cursor: pointer;
    }
  }
  .#{$provider-prefix}-badge {
    border-radius: 2px;
    font-size: 12px;
    overflow: hidden;
    padding: 6px 6px 6px 8px;
    margin: 0;
    border: none;
    border-top-right-radius: 0;
    border-bottom-right-radius: 0;
  }
}

.#{$provider-prefix}-badge {
  @extend %badge-base;
  padding: 6px;
  border: 1px solid;
  border-color: transparent;
  font-weight: $font-weight-regular;
  border-radius: 4px;
  color: $color-dark;
  margin: 2px;
  display: inline-flex;
  box-sizing: border-box;
  font-size: 12px;
  overflow: hidden;
  padding: 3px 6px;
  @each $color, $value in $theme-colors {
    @include badge-variant("-#{$color}", $value);
  }
  @each $color, $value in $theme-colors {
    @include badge-neon-variant("-neon-#{$color}", $value);
  }
}
