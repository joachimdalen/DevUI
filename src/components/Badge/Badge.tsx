import * as React from "react";
import cx from "classnames";
import { isUndefined } from "util";
import FontAwesomeIcon from "../FontAwesomeIcon/FontAwesomeIcon";
export interface BadgeProps {
  label: string;
  variant?: string;
  dismissible?: boolean;
  dismissText?: string | React.ReactNode;
  onDismiss?: () => any;
}

export class Badge extends React.PureComponent<BadgeProps> {
  static defaultProps: Partial<BadgeProps> = {
    variant: "default",
    dismissible: false
  };

  _handleClick = () => {
    const { onDismiss } = this.props;
    onDismiss && onDismiss();
  };

  public render() {
    const { label, variant, dismissible, onDismiss, dismissText } = this.props;
    const badgeClass = cx(
      "dui-badge",
      { [`dui-badge-${variant}`]: variant !== "default" },
      { [`dui-badge-dismiss`]: dismissible }
    );
    const dismissibleClass = cx("dui-badge-dismissible");
    const wrapperClass = cx("dui-badge-wrapper");
    const isDismissible = !isUndefined(onDismiss) && dismissible;
    const badgeComponent = <span className={badgeClass}>{label}</span>;
    const dismissIcon = <FontAwesomeIcon iconStyle="solid" icon="fa-times" />;
    const dismissTextComponent = isUndefined(dismissText)
      ? dismissIcon
      : dismissText;

    if (isDismissible)
      return (
        <span className={wrapperClass}>
          {badgeComponent}
          <span className={dismissibleClass} onClick={this._handleClick}>
            {dismissTextComponent}
          </span>
        </span>
      );

    return badgeComponent;
  }
}
