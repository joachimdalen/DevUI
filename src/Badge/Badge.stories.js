import React from "react";

import { storiesOf } from "@storybook/react";
import { Badge } from "./Badge";
import { text, boolean } from "@storybook/addon-knobs/react";

storiesOf("Components/Badge", module).add("Default Badge", () => (
  <Badge
    label={text("label", "Active")}
    disabled={boolean("disabled", false)}
    onClick={() => alert("hello there")}
  />
));
