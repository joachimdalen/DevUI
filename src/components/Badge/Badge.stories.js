import React from "react";

import { storiesOf } from "@storybook/react";
import { Badge } from "./Badge";
import { text, boolean } from "@storybook/addon-knobs/react";
import FontAwesomeIcon from "../FontAwesomeIcon/FontAwesomeIcon";
const variants = [
  "default",
  "primary",
  "secondary",
  "dark",
  "light",
  "danger",
  "warning",
  "blue",
  "magenta"
];
storiesOf("Components/Badge", module)
  .add("Default Badge", () => <Badge label={text("label", "Active")} />)
  .add("Dismissible", () => (
    <Badge
      label={text("label", "Active")}
      dismissible={boolean("Dismissible", true)}
      onDismiss={() => alert("Dismissed")}
    />
  ))
  .add("Custom Dismiss Icon", () => (
    <Badge
      label={text("label", "Active")}
      dismissible={boolean("Dismissible", true)}
      onDismiss={() => alert("Dismissed")}
      dismissText={<FontAwesomeIcon icon="fas fa-trash" />}
    />
  ))
  .add("Variants", () => (
    <div style={{ display: "flex", flexDirection: "column" }}>
      <div>
        {variants.map(variant => {
          return <Badge label={variant} variant={variant} />;
        })}
      </div>
      <div style={{ marginTop: "10px" }}>
        {variants.map(variant => {
          return <Badge label={variant} variant={variant} neonMode={true} />;
        })}
      </div>
    </div>
  ));
