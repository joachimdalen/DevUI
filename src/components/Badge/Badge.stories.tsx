import * as React from "react";
import { storiesOf } from "@storybook/react";
import { Badge } from "./Badge";
import { text, boolean } from "@storybook/addon-knobs/react";
import FontAwesomeIcon from "../FontAwesomeIcon/FontAwesomeIcon";
import { action } from "@storybook/addon-actions";
import variants from "../../storyUtil/variants";
storiesOf("Display Components|Badge", module)
  .add("Default Badge", () => <Badge label={text("label", "Active")} />)
  .add("Dismissible", () => (
    <Badge
      label={text("label", "Active")}
      dismissible={boolean("Dismissible", true)}
      onDismiss={action("badge-dismissed")}
    />
  ))
  .add("Custom Dismiss Icon", () => (
    <Badge
      label={text("label", "Active")}
      dismissible={boolean("Dismissible", true)}
      onDismiss={action("badge-dismissed")}
      dismissText={<FontAwesomeIcon icon="fas fa-egg" />}
    />
  ))
  .add("Variants", () => (
    <div style={{ display: "flex", flexDirection: "column" }}>
      <div>
        {variants.map(variant => {
          return <Badge label={variant} variant={variant} />;
        })}
      </div>
    </div>
  ));
