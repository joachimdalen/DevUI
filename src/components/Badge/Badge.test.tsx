import React from 'react';
import { Badge } from './Badge';
import renderer from 'react-test-renderer';

describe('Badge', () => {
    test('Badge renderes', () => {
        const component = renderer.create(
            <Badge label="Hello" />
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    test('Badge can be dismissed', () => {
        const onDismiss = jest.fn();

        const component = renderer.create(<Badge label="Hello" dismissible onDismiss={onDismiss}></Badge>)


        expect(onDismiss).toBeCalled();
    })
})