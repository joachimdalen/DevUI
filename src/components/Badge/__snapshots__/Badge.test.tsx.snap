// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge Badge renderes 1`] = `
<span
  className="dui-badge"
>
  Hello
</span>
`;
